src/components/screens/settings-screen.tsx

```diff
@@ -16,54 +16,95 @@ const SettingsScreen = () => {
   const setUsers = useSetAtom(usersAtom);
   const setError = useSetAtom(errorAtom);
 
-  const [isConsentGiven, setIsConsentGiven] = useState(
+  const [isConsentGiven, setIsConsentGiven] = useState<boolean>(
     Boolean(userProfile?.attributes?.severaUserId)
   );
-  const [isLoading, setIsLoading] = useState(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsConsentGiven(Boolean(userProfile?.attributes?.severaUserId));
+  }, [userProfile?.attributes?.severaUserId]);
 
   const handleToggleChange = () => {
-    grantSeveraOptInConsent();
+    if (isConsentGiven) {
+      revokeSeveraOptIn();
+    } else {
+      grantSeveraOptInConsent();
+    }
   };
 
   const grantSeveraOptInConsent = async () => {
     setIsLoading(true);
     try {
+      if (!usersApi) {
+        setError(strings.error.fetchFailedFlextime);
+        return;
+      }
```

Comment:
Do we really need to check if the usersApi exists? It should, so I'm not sure why this is necessary